server/tests/test_pipeline.py

```diff
@@ -285,7 +285,16 @@ def test_skip_void_parameter_from_variables_for_mongo_steps():
                 }
             }
         ]
-    ) == [{"$match": {"$and": [{"$or": [{"property2": "value2"}]}], "$ne": None}}]
+    ) == [
+        {
+            "$match": {
+                "$and": [{"$or": [{"property2": "value2"}]}, {"$nor": []}],
+                "$ne": None,
+                "$nor": [{"property9": {"$and": []}}],
+                "$or": [],
```

Comment:
Isn't this match always false ?

Reply:
and why that please ? 🤔 
This is just a dummy test on how the removal behave on empty dict/list...

Reply:
Because this would try to match an `$or` condition with an empty predicate list. If mongo's `$or` operator is defined as "at least one predicate is true", then this would always be false

Reply:
If the list is empty, it's probably because a predicate containing a `__VOID__` was removed. So it's not necessarily up to the AB

Reply:
mhhh... no 🤔 
please check again the assertion instruction, there is no `__VOID__` over there, I don't get well your point here !
```python
    assert remove_void_conditions_from_mongo_steps(
        [
            {
                "$match": {
                    "$ne": None,
                    "$and": [
                        {"$or": [{"property2": "value2"}, {"property3": {}}]},
                        {"$nor": [{"property4": {}}, {"property5": {}}]},
                    ],
                    "$or": [],
                    "$nor": [
                        {
                            "property9": {
                                "$and": [
                                    {
                                        "property7": {},
                                    }
                                ]
                            }
                        }
                    ],
                }
            }
        ]
    ) == [{"$match": {"$and": [{"$or": [{"property2": "value2"}]}], "$ne": None}}]
    ) == [
        {
            "$match": {
                "$and": [{"$or": [{"property2": "value2"}]}, {"$nor": []}],
                "$ne": None,
                "$nor": [{"property9": {"$and": []}}],
                "$or": [],
            }
        },
    ]
```

any suggestion could help 🙏🏽